bulls_and_cows/lib/api.ex

```diff
@@ -0,0 +1,34 @@
+defmodule BullsAndCows.API do
+  alias BullsAndCows
+  use GenServer
+
+  # Client
+
+  @doc """
+    Make number to guess to be picked, start the game
+  """
+  def start_link() do
+    GenServer.start_link(__MODULE__, nil)
```

Comment:
I would use the named server here.
```suggestion
    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
```